pkg/pgmodel/ingestor/insert_ctx.go

```diff
@@ -20,6 +20,12 @@ func NewWriteRequest() *prompb.WriteRequest {
 	return wrPool.Get().(*prompb.WriteRequest)
 }
 
+func NewWriteRequestWithTs(ts []prompb.TimeSeries) *prompb.WriteRequest {
+	wr := wrPool.Get().(*prompb.WriteRequest)
```

Comment:
And even better to do

```golang
// NewWriteRequest returns a new *prompb.WriteRequest from the pool and applies ts to it if ts is not nil.
func NewWriteRequest(ts []prompb.TimeSeries) *prompb.WriteRequest {
	wr := wrPool.Get().(*prompb.WriteRequest)
	if ts != nil {
		wr.Timeseries = ts
	}
	return wr
}
```

Reply:
I actually don't like that we change this like so just because of the tests. I'd prefer to add a new function (but have that function use `NewWriteRequest`) so we can use that but leave all the old code as is, using the `NewWriteRequest` without parameters.

Reply:
Since `NewWriteRequestWithTs` is only used by tests, it should be declared in a test file IMO. Less code to test, less to break.